File: solutions/Reverse-Integer/Reverse-Integer.cpp

class Solution {
public:
    int reverse(int x) {
        long long ans=0;
        while(x)
        {
            ans=ans*10+x%10;
            if(ans>INT_MAX)
                return INT_MAX;
            if(ans<INT_MIN) 
                return INT_MIN;
            x/=10;
        }
        return ans;
    }
};


 

File: solutions/Reverse-Integer/Reverse-Integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
  let ans = 0;
  while (x !== 0) {
    ans = ans * 10 + x % 10;
    if (ans > Math.pow(2, 31)) {
      return 0;
    }
    if (ans < -Math.pow(2, 31)) {
      return 0;
    }
    x = x >= 0 ? Math.floor(x / 10) : Math.ceil(x / 10);
  }
  return ans;
};